### PR TITLE
Add support for A-api-request-uuid query parameter

### PR DESCRIPTION
**Pull request by jeffersonwhite**
When the A-api-request-uuid query parameter is included in the request, it will be added to the s3 presigned url so that it can be extracted as a metric when downloaded.

### CODE REVIEW

**Comment by jeffersonwhite** [comment]
There's a corresponding rain-api-core change for this:
https://github.com/asfadmin/rain-api-core/pull/292/files

I think I will need to update the commit hashes in the dependency files in the PR once that is merged, to make sure it's getting pulled in here?

**Comment by mattp0** [comment]
@jeffersonwhite  Please see the unit test and linter issues.

**Comment by jeffersonwhite** [comment]
I've updated to address the linter findings.

For the unit tests here, they are failing because these changes depend on the changes I have in PR in rain-api-core. I'm not really sure what the process for making changes to both these repos should be, if I understand how things work I could:
- Update this PR to temporarily point the rain-api-core dependency to the branch of my forked repo so it passes the tests
- Wait until the rain-api-core PR is merged, and then update the dependency here to point to the new commit

I have run the tests locally to make sure that everything passes when all the changes are present.

File: thin_egress_app/app.py [hunk 1]
```diff
@@ -387,6 +387,16 @@ def get_user_from_token(token):
...
 
 
+@with_trace()
+def get_api_request_uuid(query_params):
+    api_request_uuid = None
+    if query_params is not None:
+        api_request_uuid = query_params.get("A-api-request-uuid")
+        if api_request_uuid is not None:
+            log.info("A-api-request-uuid query param is " + api_request_uuid)
+        return api_request_uuid
```
Comment (reweeden) [inline thin_egress_app/app.py:397]: `api_request_uuid` can just be local to the if statement.
```suggestion
    if query_params is not None:
        api_request_uuid = query_params.get("A-api-request-uuid")
        if api_request_uuid is not None:
            log.info("A-api-request-uuid query param is " + api_request_uuid)
        return api_request_uuid
        
    return None
```

**Comment by reweeden** [comment]
>     * Update this PR to temporarily point the rain-api-core dependency to the branch of my forked repo so it passes the tests

The way I've done it in the past is to point the pinned dependency version to the latest commit hash from my feature branch.

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Python
 *.pyc
 /venv
+/.venv
```
[the remark this answers is not in the document]
Comment (jeffersonwhite) [reply .gitignore]: You're right, the / isn't needed. My venv didn't automatically generate a .gitignore though, and I used the instructions in the README, so I did leave this in here (with the /)

File: requirements/requirements.in
Comment (mattp0) [inline requirements/requirements.in:1]: with core merged, I think we can move back to the commit on master

**Comment by mattp0** [comment]
Looks like code cov might of made a change that broke our overage report. I am looking into it

**Comment by mattp0** [comment]
I got the codecov working. It looks like we just need to add a test or two to cover the cases when query params are `None`
You can see the report here https://app.codecov.io/gh/asfadmin/thin-egress-app/commit/2c097a10259b1fa61f73766dd03acfff1190892e

**Review by mattp0 — APPROVED** [review]

**Review by mckadesorensen — APPROVED** [review]

**Comment by mattp0** [comment]
@jeffersonwhite Thank you for the contribution! We will start the process of testing and will keep you updated once we have a release cut for TEA.
